### PR TITLE
Implement UIs and Adding Function

### PR DESCRIPTION
**Pull request by Sandun-Induranga**
[ITP-2265]  [ITP-2268]  [ITP-2274]  [ITP-2280]  [ITP-2289]  [ITP-2283]

[ITP-2265]: https://fcodelabs.atlassian.net/browse/ITP-2265?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2268]: https://fcodelabs.atlassian.net/browse/ITP-2268?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2274]: https://fcodelabs.atlassian.net/browse/ITP-2274?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2280]: https://fcodelabs.atlassian.net/browse/ITP-2280?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2289]: https://fcodelabs.atlassian.net/browse/ITP-2289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ITP-2283]: https://fcodelabs.atlassian.net/browse/ITP-2283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/DiaryCard/DiaryCard.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          {props.name}
+        </Typography>
+        <Typography variant="body1" color={"AppWorkspace"}>
+          {isView
+            ? props.description
+            : props.description.length > 5
+            ? props.description.slice(0, 5) + "..."
+            : props.description}
+        </Typography>
+        {props.description.length > 5 ? (
```
Comment (shamal-iroshan) [inline src/components/DiaryCard/DiaryCard.tsx]: if you only need to render one component use && operator

File: src/containers/Context/Context.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Route, Routes } from "react-router-dom";
+import SignInPage from "../SignInPage/SignInPage";
+import HomePage from "../HomePage/HomePage";
+
+const Context = () => {
+  return (
+    <Routes>
+      <Route path="/" element={<SignInPage />} />
+      <Route path="/home" element={<HomePage />} />
```
Comment (shamal-iroshan) [inline src/containers/Context/Context.tsx]: use a constant file to manage routes don't hardcode them everywhere

File: src/containers/Context/Context.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { Route, Routes } from "react-router-dom";
```
Comment (shamal-iroshan) [inline src/containers/Context/Context.tsx]: what is the meaning of this context file? only pages should be in the containers directory.

File: src/containers/HomePage/HomePage.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                color="secondary"
+                sx={{ borderRadius: 10, mb: 2 }}
+                onClick={handleSubmit}
+              >
+                Submit
+              </Button>
+            </Grid>
+          </Card>
+        ) : (
+          <></>
```
Comment (shamal-iroshan) [inline src/containers/HomePage/HomePage.tsx]: here also you can use && operator

File: src/types/DiaryCardProps.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export type DiaryCardProps = {
```
Comment (shamal-iroshan) [inline src/types/DiaryCardProps.ts]: types that are needed for a page or component you can define them inside the component, specially types of props should be in the corresponding component

**Review by shamal-iroshan — CHANGES_REQUESTED** [review]
fix commented things

File: src/containers/Context/Context.tsx (hunk 3 above)
Comment (Sandun-Induranga) [reply src/containers/Context/Context.tsx]: It is the Component that manage pages according to path

**Review by shamal-iroshan — APPROVED** [review]

**Pull request merged**